tests/Kros.Ocelot.ETagCaching.Test/Policies/CacheKeyGeneratorShould.cs

```diff
@@ -0,0 +1,167 @@
+    }
+
+    [Fact]
+    public void CreateFromDownstreamRequest_ShouldNormalizeToLowerCase()
+    {
+        var request = new HttpRequestMessage()
+        {
+            Method = HttpMethod.Get,
+            RequestUri = new Uri("HTTPS://API.EXAMPLE.COM/PRODUCTS?CATEGORY=ELECTRONICS")
+        };
+        var downstreamRequest = new DownstreamRequest(request);
+
+        var result = CacheKeyGenerator.CreateFromDownstreamRequest(downstreamRequest);
+
+        result.Should().Be("get:https:api.example.com:/products:?category=electronics");
+    }
+
+    [Fact]
+    public void CreateFromUpstreamRequest_ShouldNormalizeToLowerCase()
+    {
+        var httpContext = new DefaultHttpContext();
+        httpContext.Request.Method = "GET";
+        httpContext.Request.Scheme = "HTTPS";
+        httpContext.Request.Host = new HostString("API.MYAPP.COM");
+        httpContext.Request.Path = "/API/PRODUCTS";
+        httpContext.Request.QueryString = new QueryString("?CATEGORY=ELECTRONICS");
+
+        var result = CacheKeyGenerator.CreateFromUpstreamRequest(httpContext.Request);
+
+        result.Should().Be("get:https:api.myapp.com:/api/products:?category=electronics");
+    }
+
+    [Fact]
+    public void CreateFromDownstreamRequest_ShouldHandleDifferentHttpMethods()
+    {
+        var methods = new[] { HttpMethod.Get, HttpMethod.Post, HttpMethod.Put, HttpMethod.Delete, HttpMethod.Patch };
+        var expectedMethods = new[] { "get", "post", "put", "delete", "patch" };
+
+        for (int i = 0; i < methods.Length; i++)
+        {
+            var request = new HttpRequestMessage()
+            {
+                Method = methods[i],
+                RequestUri = new Uri("https://api.example.com/test")
+            };
+            var downstreamRequest = new DownstreamRequest(request);
+
+            var result = CacheKeyGenerator.CreateFromDownstreamRequest(downstreamRequest);
+
+            result.Should().StartWith($"{expectedMethods[i]}:");
+        }
+    }
+
+    [Fact]
+    public void CreateFromUpstreamRequest_ShouldHandleDifferentHttpMethods()
+    {
+        var methods = new[] { "GET", "POST", "PUT", "DELETE", "PATCH" };
+        var expectedMethods = new[] { "get", "post", "put", "delete", "patch" };
+
+        for (int i = 0; i < methods.Length; i++)
+        {
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Method = methods[i];
+            httpContext.Request.Scheme = "https";
+            httpContext.Request.Host = new HostString("api.example.com");
+            httpContext.Request.Path = "/test";
+            httpContext.Request.QueryString = QueryString.Empty;
+
+            var result = CacheKeyGenerator.CreateFromUpstreamRequest(httpContext.Request);
+
+            result.Should().StartWith($"{expectedMethods[i]}:");
+        }
+    }
+
+    [Fact]
+    public void BothMethods_ShouldGenerateDifferentKeysForSameEndpoint()
```

Comment:
Osobne mi príde tento test úplne zbytočný.

- Keďže upstream aj downstream key testuješ na explicitné hodnoty, ktoré sú rôzne, tak nemá zmysel tieto hodnoty testovať voči sebe či sú naozaj rôzne – ak by boli náhodou rovnaké, tak to padne na jednej z tých dvoch predošlých kontrol.
- Ak zrušíme kontrolu či sú hodnoty rôzne, tak test nemá význam ako celok, pretože tie dve kontroly čo ostanú sú zahrnuté v iných testoch.